datafusion/optimizer/src/filter_null_join_keys.rs

```diff
@@ -100,11 +105,18 @@ fn create_not_null_predicate(filters: Vec<Expr>) -> Expr {
         .into_iter()
         .map(|c| Expr::IsNotNull(Box::new(c)))
         .collect();
-    // combine the IsNotNull expressions with AND
+
+    // directly unwrap since it should always have a value
     not_null_exprs
-        .iter()
-        .skip(1)
-        .fold(not_null_exprs[0].clone(), |a, b| and(a, b.clone()))
+        .into_iter()
+        .reduce(|a, b| {
+            Expr::BinaryExpr(BinaryExpr {
+                left: Box::new(a),
+                op: Operator::And,
+                right: Box::new(b),
+            })
+        })
+        .unwrap()
```

Comment:
Nice!

I think you could make this even shorter like

```suggestion
        .into_iter()
        .reduce(|a, b| a.and(b))
        .unwrap()
```

Or even use `conjunction`: https://docs.rs/datafusion/latest/datafusion/logical_expr/utils/fn.conjunction.html

```suggestion
       conjunction(not_null_exprs).unwrap()
```